dialect/sql/schema/postgres.go

```diff
@@ -526,3 +526,94 @@ func arrayType(t string) bool {
 	}
 	return true
 }
+
+func (d *Postgres) foreignKeys(ctx context.Context, tx dialect.Tx, tables []*Table) (map[*Table][]*ForeignKey, error) {
+	var fks = make(map[*Table][]*ForeignKey)
+
+	var tableLookup = make(map[string]*Table)
+
+	// TODO: include schema
+	for _, t := range tables {
+		tableLookup[t.Name] = t
+	}
+
+	for _, t := range tables {
```

Comment:
I don't follow. The query is already run once per table

Reply:
I think I understand now. I pass through all tables because I need to lookup the related columns and table to build the foreign key object

```go
&ForeignKey{
	Symbol:     constraintName,
	Columns:    []*Column{column},
	RefTable:   refTable,
	RefColumns: []*Column{refColumn}, // <--
}
```

Reply:
👍